hardware/1Wire.cpp

```diff
@@ -188,10 +188,11 @@ void C1Wire::GetDeviceDetails()
 	if ((m_devices.size() == 0) || (m_sql.m_bAcceptNewHardware))

```

Comment:
this can always be true:

m_sql.m_bAcceptNewHardware

It looks like you now are always clearing the m_devices array ?

instead of m_devices.size()==0, could you write m_devices.empty()

Maybe it should be:

`if ((m_devices.empty() == 0) && (m_sql.m_bAcceptNewHardware))`